CreatePackages.md

```diff
@@ -177,6 +177,25 @@ No matter how you decide, you are advised to state the default installation dire
 
 If you allow customizing the installation path, then append instructions on how to do that, too.
 
+### Windows environment variables
+Chocolatey installations are advised to be performed while running "as administrator".  Because of this it is important that you understand that some windows environment variables will be pinned to the administrator user and not the installation user in some circumstances.
+
+The issue can occur if a windows "standard user" installs Chocolatey packages on their system.  When the "standard user" runs "as administrator" the resulting powershell window or `cmd` window will define a subset of the environment variables to be relative to the administrator user, the shell is literally running "as administrator".
```

Comment:
Hi @ferventcoder,  thanks for responding...

Scenario is:

1. I am logged in as a user that is a "standard user" account.
![standard user](https://user-images.githubusercontent.com/14342237/41677953-9a717dee-747e-11e8-8d5d-19f4458544e2.png)
AFAIK, Windows 10 only has "Standard User" accounts or "Administrator" accounts.
2. User runs a command shell "as Administrator".
3. In that shell the environment variables listed are from the administrator account.
4. Running the installer from this command shell will install the desired software into the administrator account.

In the case of the [bug that motivated this issue](https://github.com/meteor/meteor-chocolatey-installer/issues/2), running `choco` in a non-administrator shell will fail.

Workaround for meteor is defined [here](https://github.com/meteor/meteor-chocolatey-installer/issues/2#issuecomment-341508892).

Is there another way to run as administrator?
How would UAC elevation be invoked to run the chocolatey install?

Reply:
Can we add that image?

Reply:
Can I understand if a standard user is under UAC? Or how do they actually run the command shell "as Administrator" - what does it look like from a process perspective? They simply click good to go and boom they are an admin - or do they need a login?

Reply:
I do not know what you mean "is a standard user under UAC?".

Here is how it works to set up an elevated command shell, so you can repro:
1. search for `cmd` from taskbar.
2. on the `Command Prompt` icon, right click and you will see "Run As Administrator"
3. this will invoke a UAC prompt asking for credentials for the admin account.
4. when credential are entered it will then bring up a command prompt that has administrator privileges.  Only this app will have admin privileges and any apps it calls get those privileges.  So choco can complete, but those environment variables I list are set to administrator locations, not the locations of my standard user account.

As a general rule, I have been taught to always run my windows machine from a "standard users" account.  This lessens the possibility of malware getting through (even though UAC was designed to prevent this even for administrator accounts).

Reply:
> this will invoke a UAC prompt asking for credentials for the admin account. 

MINUS credentials, this is "is a standard user under UAC?"

Reply:
So there is a difference between elevation for an admin user (regular UAC) in that it just stops and asks if you want to elevate without needing credentials. So you see why I need this to be VERY explicit that the process where you actually enter OTHER credentials, you will be running under the context of that OTHER user and all things will be as if you were logged in as that user (because in effect you are).

Reply:
An administrative user who is not elevated has a similar experience:

1. search for cmd from taskbar. (or PowerShell)
1. on the `Command Prompt` icon, right click and you will see "Run As Administrator"
1. this will invoke a UAC prompt asking the user to confirm.
1. when confirmed, it will then bring up a command prompt that has administrator privileges - elevating that admin user that is subject to UAC.

See how this is dangerously close to being confused with the scenario you described?

>Here is how it works to set up an elevated command shell, so you can repro:
> 1. search for `cmd` from taskbar.
> 2. on the `Command Prompt` icon, right click and you will see "Run As Administrator"
> 3. this will invoke a UAC prompt asking for credentials for the admin account.
> 4. when credential are entered it will then bring up a command prompt that has administrator privileges.

Reply:
The only difference between the two scenarios is that with your process:

* Credentials for a separate account must be entered
* That process runs under an entirely different account

Reply:
The changes I made are instructions for package authors, not end users.  So I was not thinking that I was instructing end users how to elevate.  What exactly do you want me to change? &/or where were you tripped up?

>Because of this it is important that you understand that some windows environment variables will be pinned to the administrator user and not the installation user in some circumstances.

-- this explains that there are some circumstances (not all) that can cause problems.  If the user is installing for the admin from within an admin account all works as expected.

>The issue can occur if a windows "standard user" installs Chocolatey packages on their system.  When the "standard user" runs "as administrator" the resulting powershell window or `cmd` window will define a subset of the environment variables to be relative to the administrator user, the shell is literally running "as administrator".

-- then I define the specific circumstance.  I can elaborate more on how a user with a standard account elevates to administrator?

>The only difference between the two scenarios is that with your process:
> - Credentials for a separate account must be entered
> - That process runs under an entirely different account

Indeed, and that is the scenario that breaks.  But this information is for package writers to think more about the usage of the environment variables in their scripts.

Is there another way a standard user could elevate privileges without the environment variables being set to the admin user?

Reply:
Please elaborate more on what a standard user is - remember not everyone is super familiar with the types of users - some package maintainers are vendors who make cross platform tools and are not even going to understand "standard user", let alone too much about Windows itself. You can almost never go wrong with images and explaining it VERY explicitly. 👍

Reply:
The answer for your specific scenario is likely self-service in C4B.

Reply:
C4B?

Reply:
OK, will do some more elaboration.  Thanks!

Reply:
Chocolatey for Business - commercial edition of Chocolatey. It has a self-service installation concept that allows non-administrators to be able to install, upgrade, etc software. Your specific use case may not be supported well by Chocolatey open source 

It's not going to be possible for a package creator to know that you were a different user, because choco is run from administrator, and by the time it gets to them, still same. No context that this came from a user that logged into a process as a different user (that's essentially what run as does). The best you can do here is document what YOU the user would need to do to update your environment variables (perhaps a script you could run once you got into that shell) that would set everything the way you wanted BEFORE you execute choco. HTH